src/otx/v2/adapters/torch/dataset.py

```diff
@@ -0,0 +1,210 @@
+
+        dataset_cfg = dataset.configs if hasattr(dataset, "configs") else dataset
+        dataloader.configs = {
+            "batch_size": batch_size,
+            "sampler": sampler_cfg,
+            "num_workers": num_workers,
+            "pin_memory": pin_memory,
+            "shuffle": shuffle,
+            "persistent_workers": persistent_workers,
+            "dataset": dataset_cfg,
+            **kwargs,
+        }
+        return dataloader
+
+    def subset_dataloader(
+        self,
+        subset: str,
+        pipeline: dict | list | None = None,
+        batch_size: int | None = None,
+        num_workers: int | None = None,
+        config: str | dict | None = None,
+        shuffle: bool = True,
+        pin_memory: bool = False,
+        drop_last: bool = False,
+        sampler: Sampler | (Iterable | dict) | None = None,
+        persistent_workers: bool = False,
+        **kwargs,
+    ) -> TorchDataLoader:
+        r"""Torch Based Dataset.subset_dataloader.
+
+        Args:
+            subset (str): Enter an available subset of that dataset.
+            pipeline (Optional[Union[list, dict]], optional):
+                Dataset Pipeline. Defaults to None.
+            batch_size (Optional[int], optional): How many samples per batch to load. Defaults to None.
+            num_workers (Optional[int], optional): How many subprocesses to use for data loading.
+                ``0`` means that the data will be loaded in the main process. Defaults to None.
+            config (Optional[Union[str, dict]], optional): Path to configuration file or Config.
+                Defaults to None.
+            shuffle (bool, optional): Set to ``True`` to have the data reshuffled at every epoch. Defaults to True.
+            pin_memory (bool, optional): If ``True``, the data loader will copy Tensors
+                into device/CUDA pinned memory before returning them.  If your data elements
+                are a custom type, or your :attr:`collate_fn` returns a batch that is a custom type,
+                see the example below. Defaults to False.
+            drop_last (bool, optional): value for whether to drop the last data when the batch is not divided up.
+                Defaults to False.
+            sampler (Optional[Union[Sampler, Iterable, Dict]], optional): Defines the strategy to draw
+                samples from the dataset. Can be any ``Iterable`` with ``__len__``
+                implemented. If specified, :attr:`shuffle` must not be specified.. Defaults to None.
+            persistent_workers (bool, optional): If ``True``, the data loader will not shutdown
+                the worker processes after a dataset has been consumed once. This allows to
+                maintain the workers `Dataset` instances alive. Defaults to False.
+            **kwargs (Any): Additional arguments to pass to the DataLoader constructor.
+
```

Comment:
Also would it be possible to add an `Example` section here?

Reply:
I also added an Example section.